Source/i_flmusic.c

```diff
@@ -47,9 +49,51 @@ static void FL_Mix_Callback(void *udata, Uint8 *stream, int len)
     }
 }
 
+static void *FL_sfopen(const char *lumpname)
+{
+    MEMFILE *instream;
+    int lumpnum = W_GetNumForName(lumpname);
+    int len = W_LumpLength(lumpnum);
+    void *data = W_CacheLumpNum(lumpnum, PU_STATIC);
+
+    instream = mem_fopen_read(data, len);
+
+    return instream;
+}
+
+static int FL_sfread(void *buf, fluid_long_long_t count, void *handle)
+{
+    if (mem_fread(buf, sizeof(byte), count, (MEMFILE *)handle) == count)
+    {
+        return FLUID_OK;
+    }
+    return FLUID_FAILED;
+}
+
+static int FL_sfseek(void *handle, fluid_long_long_t offset, int origin)
+{
+    if (mem_fseek((MEMFILE *)handle, offset, origin) < 0)
+    {
+        return FLUID_FAILED;
+    }
+    return FLUID_OK;
+}
+
+static int FL_sfclose(void *handle)
+{
+    mem_fclose((MEMFILE *)handle);
```

Comment:
Should we free the lump as well? I mean set to `PU_CACHE`?

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/454/commits/45000071d4836580c92228961ea750fc52116437